core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -2479,4 +2479,14 @@ package object config {
       .version("3.4.0")
       .booleanConf
       .createWithDefault(false)
+
+  private[spark] val STAGE_MAX_ATTEMPTS =
+    ConfigBuilder("spark.stage.maxAttempts")
+      .doc("The max attempts for a stage, the spark job will be aborted if any of its stages is " +
+        "resubmitted multiple times beyond the limitation. The value should be no less " +
+        "than `spark.stage.maxConsecutiveAttempts` which defines the max attempts for " +
+        "fetch failures.")
```

Comment:
I did not notice this earlier.
We should rephrase this - we use the max of `spark.stage.maxAttempts` and `spark.stage.maxConsecutiveAttempts`.
So if user specifies a value lower than `spark.stage.maxConsecutiveAttempts`, we would use `spark.stage.maxConsecutiveAttempts` - which is not what the description is conveying.

Something like:

```suggestion
      .doc("Specify the max attempts for a stage - the spark job will be aborted if any of its stages is " +
        "resubmitted multiple times beyond the limitation. The maximum number of stage retries is the maximum of " +
        "`spark.stage.maxConsecutiveAttempts` and `spark.stage.maxAttempts`")
```